STYLEGUIDE.md

```diff
@@ -424,6 +487,92 @@ Sets or overrides the provider for MyService to the myFactory factory function.
 ### Useful links:
 - #### Docs: https://angular.io/guide/dependency-injection
 
+## COMPONENT COMMUNICATION
+
+Unidirectional data flow is a major concept in modern JS frameworks.
+Components are categorized as either smart components or presentation components. Smart components should interact with services/stores and pass data to presentation components. If events from presentation components need to be communicated to other presentation components, the preferred route is to communicate to the parent smart component using a service.
+
+This may seem like a trivial example that could be achieved with inputs and outputs, but as your component hierarchy grows, it becomes cumbersome to handle passing data up and down components.
+
+Service: Holds a subject that the parent smart component will create a subscription and the presentation components will publish an event. 
+
+Best practices:
+
+- only expose subjects as observables, this allows the implementation details to only be defined in the service. 
+- end observable names with a $ (i.e., myObservable$), easily identifies a property as an observable
+
+```js
+export class MyService {
+  private eventSubject: Subject<{message: string}> = new Subject();
+
+  constructor() {}
+  
+  get event$(): Observable<{message: string}> {
+    this.eventSubject.asObservable();
+  }
+  
+  publishEvent(event): {message: string} {
+    return this.eventSubject.next(event);
+  }
+}
+```
+
+Smart Component: The smart component sets the observable from MyService as a public instance variable to be used in the template.
+
+```js
+export class SmartComponent {
+  public event$: Observable<{message: string} = this.myService.event$;
+
+  constructor(
+    private myService: MyService
+  ) {}
+}
+
+```
+
+The smart component's template is composed of 2 presentation components: one to emit an event to the service and one to receive the event from the smart component.
+
+We are using the async pipe here to manage creating/destroying the subscription, as well as calling change detection and passing down the event to the presentation-bar component.
+
+```html
+<presentation-foo></presentation-foo>
+<presentation-bar [event]="event$ | async"></presentation-bar>
+```
+
+Presentation Foo component: This presentation component is emitting an event that the other presentation components needs to listen to.
+
+```js
+export class PresentationFooComponent {
+  constructor(
+    private myService: MyService
+  ) {}
+  
+  // imagine a click event on button element
+  onButtonClick() {
+    this.myService.publishEvent({ event: 'Hello' });
+  }
+}
+```
+Presentation Bar component: This presentation component is getting passed the latest value from the eventSubject by the smart components subscription to the event$ observable. 
+
+```js
+export class PresentationFooComponent {
```

Comment:
@guyjstitt - Thanks for all those additions. This is just a typo here I guess - Should be PresentationBarComponent

Reply:
Thanks for catching that! Fixed plus a couple other minor typos.